### ISSUE TITLE
Consecutive plots are automatically reported as a single plot

### ISSUE THREAD

**Issue by yoavhalevi** [allegroai/trains#140]
Hey,

I have a loop that plots 4 graphs in each iteration and saves them to disk with `matplotlib`, after each plot I make sure to close the figure and the axes. the plots are automatically reported, even though I only create them in order to save them. plus, the 4 plots from the same loop iteration are concatenated as a single large plot and then reported, causing a very long reporting time and a warning by `plotly`:
```
UserWarning: figure embedded in figure: something is wrong
UserWarning: axes embedded in axes: something is wrong
```
sometime, like in the following example, the plots are not shown appropriately while they have a spot allocated in the dashboard. (if you look closely you'll see the screen is divided by 4, corresponding for the 4 plots in that loop iteration)
![image](https://user-images.githubusercontent.com/59438344/83730169-5cefc280-a651-11ea-908e-8fa7bb304b03.png)

1. Is it possible to turn off automatic report when using `plt.plot()`?
2. how should I overcome the concatenation of the plots?

Thanks

**Comment by bmartinn** [comment]
Hi @yoavhalevi ,
The matplotlib automagic will catch any plot show/save and store it into *trains-server*.
Seems like it is missing a few of the plots (I guess that's the reason for the empty 4 plot screenshot)
What's your `trains` version and what's the `matplotlib` version you are using?
Any chance you can provide a short code to reproduce the issue? 

Regrading your questions:
1. Yes you can disable matplotlib auto connect magic, see auto_connect_frameworks in [documentation](https://allegro.ai/docs/task.html#trains.task.Task.init):
```python
task = Task.init('examples', 'disable matplotlib connect', auto_connect_frameworks={'matplotlib': False})
```
2. Any code to reproduce the plot concatenation will be appreciated ;)

**Comment by yoavhalevi** [comment]
Hey @bmartinn ,

Thank you for your reply.
I am using `trains` version `0.14.2rc0`. using demo `trains-server` for running the following code.

1. I will definitely try that - thanks!
2. I wrote a small script that imitates my use case. while running it I encountered:
    - plots were concatenated, although it seems like my first observation was wrong and the concatenation isnt for plots from the same loop iteration but some other re-use mechanism of the placeholder for plots in the dashboard.
    - plots aren't displayed (missing).
    - the warning messages from `plotly` appear.

```
import numpy as np
import matplotlib.pyplot as plt
from sklearn.metrics import ConfusionMatrixDisplay

from trains import Task
task = Task.init(project_name="trains_bug_test", task_name="concatenated_plots")
trains_logger = Task.current_task().get_logger()

class Test():
    
    def __init__(self):
        self.num_classes = 2
        self.class_names = ["class {}".format(i) for i in range(self.num_classes)]

    def generate_dummy_cm(self, index):
        num_classes = self.num_classes
        cm = np.arange(num_classes*num_classes).reshape((num_classes, num_classes))
        cm += index
        return cm

    def plot_dummy_cm(self, cm, index, run_id):
        cmd = ConfusionMatrixDisplay(cm.astype(np.int), display_labels=self.class_names)
        cmd.plot()
        i, j = run_id
        filename="plot_cm_{}_run_{},{}".format(index, i, j)
        plt.savefig(filename)
        plt.cla()
        plt.clf()
        plt.close('all')
        
    def run(self):
        for j in range(2):
            for i in np.arange(100):
                cm_1 = self.generate_dummy_cm(1)
                self.plot_dummy_cm(cm_1, index=1, run_id=(i,j))
                cm_2 = self.generate_dummy_cm(2)
                self.plot_dummy_cm(cm_2, index=2, run_id=(i,j))
                cm_3 = self.generate_dummy_cm(3)
                self.plot_dummy_cm(cm_3, index=3, run_id=(i,j))
                cm_4 = self.generate_dummy_cm(4)
                self.plot_dummy_cm(cm_4, index=4, run_id=(i,j))

test = Test()
test.run()
```

**Comment by bmartinn** [comment]
Thanks @yoavhalevi for the code sample! 
I'll check if I can reproduce the behavior and report back on my findings.
At least regrading the plotly warning, they were fixed in v0.15.0 see #139 :)

**Comment by bmartinn** [comment]
Hi @yoavhalevi quick update.
Thank you again for the reproducing code, it was a breeze to debug :)

It seems the problem is actually Web UI related only, and the guys found a weird way to overcome it, you zoom in then you zoom out in the browser, and voila the plot is there :smile: 

This bug will be fixed in 0.15.1, which is due to be released in the next two weeks or so.
There is also a fix to the plot zoom issue (sometimes when you select the zoom area, it would zoom to a different area on the plot, anyhow that is fixed as well)

I'll update here once *trains-server* v0.15.1 is out with a fix.

**Comment by bmartinn** [comment]
Hi @yoavhalevi 
I forgot to update, *trains-server* 0.15.1 is out with a fix to the matplotlib bug :)
Hurray! :confetti_ball:
